enumerable.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+# rubocop:disable all
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield self[i]
+        i += 1
+      end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+    array2 = []
+      i = 0
+      while i < size
+        array2 << self[i] if yield self[i]
+        i += 1
+      end
+    array2
+  end
+
+  def my_all?(type = nil)
+    ary = []
+    has = {}
+    if block_given?
+        if self.is_a?(Array)
+            self.my_each { |n| yield(n) == true ? ary.push(true) : false }
+            self.length == ary.length ? (puts "true") : (puts "false")
+            ary
+        else
+            self.my_each { |k, v| yield(k, v) == true ? has[k] = v : (puts "hash") }
+            has
+        end
+    
+    elsif type != nil
+        self.my_each { |n| n.is_a?(type) ? ary.push(true) : false }
+        self.length == ary.length ? (puts "true") : (puts "false")
+    
+    else
+        self.my_each { |n| n != false && n != nil ? ary.push(true) : false }
+        self.length == ary.length ? (puts "true") : (puts "false")
+    end
+  end
```

Comment:
```ruby
irb(main):013:0> array
=> [0, 5, 3, 1, 1, 8, 0, 2, 0, 7, 5, 2, 7, 8, 6, 5, 8, 5, 2, 2, 6, 1, 3, 7, 5, 3, 0, 2, 2, 6, 4, 4, 0, 7, 6, 1, 8, 6, 3, 6, 4, 1, 6, 6, 1, 4, 8, 7, 1, 8, 2, 0, 0, 4, 8, 4, 2, 3, 5, 3, 2, 6, 2, 6, 8, 2, 3, 2, 4, 6, 7, 5, 8, 3, 2, 2, 2, 0, 4, 2, 0, 7, 5, 6, 6, 3, 2, 3, 8, 4, 6, 5, 4, 4, 7, 8, 7, 6, 2, 8]
irb(main):014:0> true_block = proc { |num| num <= 9 }
=> #<Proc:0x000055b36ae9ad90@(irb):14>
irb(main):017:0> array.all?(&true_block)
=> true
irb(main):018:0> array.my_all?(&true_block)
true
=> [true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true]
```

Reply:
```ruby
irb(main):022:0> true_array = [1, true, 'hi', []]
=> [1, true, "hi", []]
irb(main):023:0> true_array.all?
=> true
irb(main):024:0> true_array.my_all?
true
=> nil
```

Reply:
```ruby
irb(main):027:0> array
=> [0, 5, 3, 1, 1, 8, 0, 2, 0, 7, 5, 2, 7, 8, 6, 5, 8, 5, 2, 2, 6, 1, 3, 7, 5, 3, 0, 2, 2, 6, 4, 4, 0, 7, 6, 1, 8, 6, 3, 6, 4, 1, 6, 6, 1, 4, 8, 7, 1, 8, 2, 0, 0, 4, 8, 4, 2, 3, 5, 3, 2, 6, 2, 6, 8, 2, 3, 2, 4, 6, 7, 5, 8, 3, 2, 2, 2, 0, 4, 2, 0, 7, 5, 6, 6, 3, 2, 3, 8, 4, 6, 5, 4, 4, 7, 8, 7, 6, 2, 8]
irb(main):028:0> array.all?(Integer)
=> true
irb(main):029:0> array.my_all?(Integer)
true
=> nil
```

Reply:
```ruby
irb(main):033:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):034:0> words.my_all?(/d/)
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):34
        4: from enumerable.rb:47:in `my_all?'
        3: from enumerable.rb:8:in `my_each'
        2: from enumerable.rb:47:in `block in my_all?'
        1: from enumerable.rb:47:in `is_a?'
TypeError (class or module required)
irb(main):035:0> words.all?(/d/)
=> true
irb(main):036:0>
```

Reply:
```ruby
irb(main):039:0> array
=> [0, 5, 3, 1, 1, 8, 0, 2, 0, 7, 5, 2, 7, 8, 6, 5, 8, 5, 2, 2, 6, 1, 3, 7, 5, 3, 0, 2, 2, 6, 4, 4, 0, 7, 6, 1, 8, 6, 3, 6, 4, 1, 6, 6, 1, 4, 8, 7, 1, 8, 2, 0, 0, 4, 8, 4, 2, 3, 5, 3, 2, 6, 2, 6, 8, 2, 3, 2, 4, 6, 7, 5, 8, 3, 2, 2, 2, 0, 4, 2, 0, 7, 5, 6, 6, 3, 2, 3, 8, 4, 6, 5, 4, 4, 7, 8, 7, 6, 2, 8]
irb(main):040:0> array.all?(3)
=> false
irb(main):041:0> array.my_all?(3)
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):41
        4: from enumerable.rb:47:in `my_all?'
        3: from enumerable.rb:8:in `my_each'
        2: from enumerable.rb:47:in `block in my_all?'
        1: from enumerable.rb:47:in `is_a?'
TypeError (class or module required)
irb(main):042:0> 
```

Reply:
Please `don't print the answer`, **return it**

Check the comment and compare the two result